src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -169,13 +172,16 @@ def _prepare_mgmt_client_kwargs_track2(cli_ctx, cred):
 
     # Enable CAE support in mgmt SDK
     from azure.core.pipeline.policies import BearerTokenCredentialPolicy
+    from azure.cli.core.util import get_http_transport
 
     # Track 2 SDK maintains `scopes` and passes `scopes` to get_token.
     scopes = resource_to_scopes(cli_ctx.cloud.endpoints.active_directory_resource_id)
     policy = BearerTokenCredentialPolicy(cred, *scopes)
+    transport = get_http_transport()
 
     client_kwargs['credential_scopes'] = scopes
     client_kwargs['authentication_policy'] = policy
+    client_kwargs['transport'] = transport
```

Comment:
`transport` should have already been set by `_prepare_client_kwargs_track2`.

Reply:
Exactly. How could I miss that. 🥲